lib/open_telemetry_exporter/logger.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "logger"
+
+module OpenTelemetryExporter
+  class Logger
+    class << self
+      def increment(name, tags: {}, by: 1)
+        if OpenTelemetryExporter.configuration.opentelemetry_enabled?
+          begin
+            meter = OpenTelemetryExporter.configuration.opentelemetry_meter
+            raise "OpenTelemetry meter not configured" unless meter
+
+            meter.create_counter(name).add(by, attributes: convert_tags(tags))
+            by
+          rescue => e
+            log_error("OpenTelemetry error for metric '#{name}': #{e.message}")
+            log_error("Falling back to configured fallback")
+            call_fallback_increment(name, tags: tags, by: by)
+          end
+        else
+          call_fallback_increment(name, tags: tags, by: by)
+        end
+      end
+
+      def timing(name, value, tags: {})
+        if OpenTelemetryExporter.configuration.opentelemetry_enabled?
+          begin
+            meter = OpenTelemetryExporter.configuration.opentelemetry_meter
+            raise "OpenTelemetry meter not configured" unless meter
+
+            meter.create_histogram(name).record(value, attributes: convert_tags(tags))
+            value
+          rescue => e
+            log_error("OpenTelemetry error for timing metric '#{name}': #{e.message}")
+            log_error("Falling back to configured fallback")
+            call_fallback_timing(name, value, tags: tags)
+          end
+        else
+          call_fallback_timing(name, value, tags: tags)
+        end
+      end
+
+      def histogram(name, value, tags: {})
+        if OpenTelemetryExporter.configuration.opentelemetry_enabled?
+          begin
+            meter = OpenTelemetryExporter.configuration.opentelemetry_meter
+            raise "OpenTelemetry meter not configured" unless meter
+
+            meter.create_histogram(name).record(value, attributes: convert_tags(tags))
+            value
+          rescue => e
+            log_error("OpenTelemetry error for histogram metric '#{name}': #{e.message}")
+            log_error("Falling back to configured fallback")
+            call_fallback_histogram(name, value, tags: tags)
+          end
+        else
+          call_fallback_histogram(name, value, tags: tags)
+        end
+      end
+
+      private
+
+      def call_fallback_increment(name, tags: {}, by: 1)
+        fallback_object = OpenTelemetryExporter.configuration.fallback_object
+        
+        unless fallback_object
+          return by
+        end
+
+        if defined?(Datadog::Statsd) && fallback_object.is_a?(Datadog::Statsd)
```

Comment:
There's a lot of duplication, and if this is meant to be "extensible" (to other libraries), it shouldn't be hard-coded to `Datadog::Statsd`. Feels like these belong in some `OpenTelemetryExporter::DataDogAdapter` or something like that, with some sort of factory method returning whatever the configured (or default) fallback adapter is.